klient/conf/main_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conf
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"k8s.io/client-go/util/homedir"
+)
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	teardown()
+	os.Exit(code)
+}
+
+func setup() {
+	home := homedir.HomeDir()
+
+	path := filepath.Join(home, "test", ".kube", "config")
+
+	// check if file exists
+	_, err := os.Stat(path)
+	// create file if not exists
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(home+"/test/.kube", 0777)
+		if err != nil {
+			fmt.Println("failed to create .kube dir", err)
+			return
+		}
+
+		// generate kube config data
+		data := genKubeconfig("test-context")
+
+		err = createFile(path, data)
+		if err != nil {
+			fmt.Println("failed to create config file", err)
+			return
+		}
+	}
+
+	fmt.Println("file created successfully", path)
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "Paths to a kubeconfig. Only required if out-of-cluster.")
+
+	// set --kubeconfig flag
+	err = flag.Set("kubeconfig", path)
+	if err != nil {
+		fmt.Println("unexpected error while setting flag value", err)
+		return
+	}
+
+	flag.Parse()
+}
+
+func createFile(path, data string) error {
+	return ioutil.WriteFile(path, []byte(data), 0644)
+}
+
+// genKubeconfig used to genearte kube config file
+// we can provide multiple contexts as well
+func genKubeconfig(contexts ...string) string {
```

Comment:
Wondering if we should use the real type representing kubeconfig and json marshal it instead.

Reply:
(I realize this is just for tests so maybe not a big deal)